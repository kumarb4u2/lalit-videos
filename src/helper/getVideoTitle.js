const getVideoTitle = item => {
  const splitTitle = item.split('___');
  if (splitTitle.length > 1) {
    const titleWithExtension = splitTitle[splitTitle.length - 1];
    const titleWithoutExtension = titleWithExtension.split('.');
    if (titleWithoutExtension.length > 1) {
      titleWithoutExtension.pop();
      return titleWithoutExtension.join('.');
    }
  }
  return item;
};

export default getVideoTitle;
