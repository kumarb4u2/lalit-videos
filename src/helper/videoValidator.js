const isValidVideoFile = type => {
  const validFileType = [
    'video/x-flv',
    'video/mp4',
    'application/x-mpegURL',
    'video/MP2T',
    'video/3gpp',
    'video/quicktime',
    'video/x-msvideo',
    'video/x-ms-wmv'
  ];
  return validFileType.indexOf(type) >= 0;
};

export default isValidVideoFile;
