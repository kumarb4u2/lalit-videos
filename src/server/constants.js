const STATIC_PATH = './src/server/public';
const UPLOAD_DIRECTORY = 'uploads';

module.exports = {
  STATIC_PATH,
  UPLOAD_DIRECTORY
};
