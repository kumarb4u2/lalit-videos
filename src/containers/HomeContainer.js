import React, { Component } from 'react';
import axios from 'axios';
import { withRouter, Link } from 'react-router-dom';
import '../App.css';
import { BASE_URL } from '../helper/constants';
import SingleVideo from '../components/SingleVideo';
import getVideoTitle from '../helper/getVideoTitle';

// export to test unwrapped component
export class HomeContainer extends Component {
  state = {
    videos: []
  };
  componentDidMount() {
    axios.get(`${BASE_URL}/videos`).then(response => {
      this.setState({ videos: response.data.videos });
    });
  }

  playVideo = title => {
    this.props.history.push(`/play/${title}`);
  };

  render() {
    const { videos } = this.state;
    return (
      <div className="container-fluid container-padding">
        {videos.length === 0 && (
          <div className="jumbotron">
            <h1 className="display-4">Hello!</h1>
            <p className="lead">No video is present in library.</p>
            <hr className="my-4" />
            <p>Click Upload button to upload a video.</p>
            <Link className="btn btn-warning btn-lg" to="/upload" role="button">
              Upload
            </Link>
          </div>
        )}
        {videos.length > 0 && (
          <div className="row">
            {this.state.videos.map(item => (
              <div key={item} className="col-xs-12 col-sm-4 video-margin">
                <SingleVideo
                  src={`${BASE_URL}/uploads/${item}`}
                  title={getVideoTitle(item)}
                  fullName={item}
                  onClickHandler={this.playVideo}
                />
              </div>
            ))}
          </div>
        )}
      </div>
    );
  }
}

export default withRouter(HomeContainer);
