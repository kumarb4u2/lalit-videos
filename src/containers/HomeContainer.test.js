import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import HomeContainer from './HomeContainer';
import { BASE_URL } from '../helper/constants';

jest.mock('axios', () => ({
  get: jest.fn(() =>
    Promise.resolve({
      data: {
        videos: [
          '1549022169094.mp4',
          '1549022172369.mp4',
          '1549022175248.mp4',
          '1549058357078.mp4',
          '1549068900721.mp4',
          '1549070037731.mp4'
        ]
      }
    })
  )
}));

describe('HomeContainer', () => {
  it('should match snapshot in default state', () => {
    const component = shallow(<HomeContainer />);
    expect(component).toMatchSnapshot();
  });

  it('should get all videos on load', () => {
    shallow(<HomeContainer />);
    expect(axios.get).toBeCalledWith(`${BASE_URL}/videos`);
  });
});
