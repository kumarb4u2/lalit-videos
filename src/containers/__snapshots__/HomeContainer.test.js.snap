// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should match snapshot if there are some videos present in state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="row"
        >
          <div
            className="col-xs-12 col-sm-4 video-margin"
          >
            <SingleVideo
              fullName="1233___some.mp4"
              onClickHandler={[Function]}
              src="http://localhost:4000/uploads/1233___some.mp4"
              title="some"
            />
          </div>
          <div
            className="col-xs-12 col-sm-4 video-margin"
          >
            <SingleVideo
              fullName="2345___any.mp4"
              onClickHandler={[Function]}
              src="http://localhost:4000/uploads/2345___any.mp4"
              title="any"
            />
          </div>
        </div>,
      ],
      "className": "container-fluid container-padding",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs-12 col-sm-4 video-margin"
            >
              <SingleVideo
                fullName="1233___some.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/1233___some.mp4"
                title="some"
              />
            </div>,
            <div
              className="col-xs-12 col-sm-4 video-margin"
            >
              <SingleVideo
                fullName="2345___any.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/2345___any.mp4"
                title="any"
              />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1233___some.mp4",
            "nodeType": "host",
            "props": Object {
              "children": <SingleVideo
                fullName="1233___some.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/1233___some.mp4"
                title="some"
              />,
              "className": "col-xs-12 col-sm-4 video-margin",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fullName": "1233___some.mp4",
                "onClickHandler": [Function],
                "src": "http://localhost:4000/uploads/1233___some.mp4",
                "title": "some",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2345___any.mp4",
            "nodeType": "host",
            "props": Object {
              "children": <SingleVideo
                fullName="2345___any.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/2345___any.mp4"
                title="any"
              />,
              "className": "col-xs-12 col-sm-4 video-margin",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fullName": "2345___any.mp4",
                "onClickHandler": [Function],
                "src": "http://localhost:4000/uploads/2345___any.mp4",
                "title": "any",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="row"
          >
            <div
              className="col-xs-12 col-sm-4 video-margin"
            >
              <SingleVideo
                fullName="1233___some.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/1233___some.mp4"
                title="some"
              />
            </div>
            <div
              className="col-xs-12 col-sm-4 video-margin"
            >
              <SingleVideo
                fullName="2345___any.mp4"
                onClickHandler={[Function]}
                src="http://localhost:4000/uploads/2345___any.mp4"
                title="any"
              />
            </div>
          </div>,
        ],
        "className": "container-fluid container-padding",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs-12 col-sm-4 video-margin"
              >
                <SingleVideo
                  fullName="1233___some.mp4"
                  onClickHandler={[Function]}
                  src="http://localhost:4000/uploads/1233___some.mp4"
                  title="some"
                />
              </div>,
              <div
                className="col-xs-12 col-sm-4 video-margin"
              >
                <SingleVideo
                  fullName="2345___any.mp4"
                  onClickHandler={[Function]}
                  src="http://localhost:4000/uploads/2345___any.mp4"
                  title="any"
                />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1233___some.mp4",
              "nodeType": "host",
              "props": Object {
                "children": <SingleVideo
                  fullName="1233___some.mp4"
                  onClickHandler={[Function]}
                  src="http://localhost:4000/uploads/1233___some.mp4"
                  title="some"
                />,
                "className": "col-xs-12 col-sm-4 video-margin",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "fullName": "1233___some.mp4",
                  "onClickHandler": [Function],
                  "src": "http://localhost:4000/uploads/1233___some.mp4",
                  "title": "some",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2345___any.mp4",
              "nodeType": "host",
              "props": Object {
                "children": <SingleVideo
                  fullName="2345___any.mp4"
                  onClickHandler={[Function]}
                  src="http://localhost:4000/uploads/2345___any.mp4"
                  title="any"
                />,
                "className": "col-xs-12 col-sm-4 video-margin",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "fullName": "2345___any.mp4",
                  "onClickHandler": [Function],
                  "src": "http://localhost:4000/uploads/2345___any.mp4",
                  "title": "any",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`HomeContainer should match snapshot in default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="jumbotron"
        >
          <h1
            className="display-4"
          >
            Hello!
          </h1>
          <p
            className="lead"
          >
            No video is present in library.
          </p>
          <hr
            className="my-4"
          />
          <p>
            Click Upload button to upload a video.
          </p>
          <Link
            className="btn btn-warning btn-lg"
            replace={false}
            role="button"
            to="/upload"
          >
            Upload
          </Link>
        </div>,
        false,
      ],
      "className": "container-fluid container-padding",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="display-4"
            >
              Hello!
            </h1>,
            <p
              className="lead"
            >
              No video is present in library.
            </p>,
            <hr
              className="my-4"
            />,
            <p>
              Click Upload button to upload a video.
            </p>,
            <Link
              className="btn btn-warning btn-lg"
              replace={false}
              role="button"
              to="/upload"
            >
              Upload
            </Link>,
          ],
          "className": "jumbotron",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello!",
              "className": "display-4",
            },
            "ref": null,
            "rendered": "Hello!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No video is present in library.",
              "className": "lead",
            },
            "ref": null,
            "rendered": "No video is present in library.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "my-4",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click Upload button to upload a video.",
            },
            "ref": null,
            "rendered": "Click Upload button to upload a video.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Upload",
              "className": "btn btn-warning btn-lg",
              "replace": false,
              "role": "button",
              "to": "/upload",
            },
            "ref": null,
            "rendered": "Upload",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="jumbotron"
          >
            <h1
              className="display-4"
            >
              Hello!
            </h1>
            <p
              className="lead"
            >
              No video is present in library.
            </p>
            <hr
              className="my-4"
            />
            <p>
              Click Upload button to upload a video.
            </p>
            <Link
              className="btn btn-warning btn-lg"
              replace={false}
              role="button"
              to="/upload"
            >
              Upload
            </Link>
          </div>,
          false,
        ],
        "className": "container-fluid container-padding",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="display-4"
              >
                Hello!
              </h1>,
              <p
                className="lead"
              >
                No video is present in library.
              </p>,
              <hr
                className="my-4"
              />,
              <p>
                Click Upload button to upload a video.
              </p>,
              <Link
                className="btn btn-warning btn-lg"
                replace={false}
                role="button"
                to="/upload"
              >
                Upload
              </Link>,
            ],
            "className": "jumbotron",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello!",
                "className": "display-4",
              },
              "ref": null,
              "rendered": "Hello!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No video is present in library.",
                "className": "lead",
              },
              "ref": null,
              "rendered": "No video is present in library.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "my-4",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click Upload button to upload a video.",
              },
              "ref": null,
              "rendered": "Click Upload button to upload a video.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Upload",
                "className": "btn btn-warning btn-lg",
                "replace": false,
                "role": "button",
                "to": "/upload",
              },
              "ref": null,
              "rendered": "Upload",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
